src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1546,10 +1547,27 @@ public CollectionName GetCollection(string collection, bool throwIfDoesNotExist)
             return collectionName;
         }
 
-        public CollectionName ExtractCollectionName(DocumentsOperationContext context, BlittableJsonReaderObject document)
+        public CollectionName ExtractCollectionName(DocumentsOperationContext context, ref BlittableJsonReaderObject document)
         {
-            var originalCollectionName = CollectionName.GetCollectionName(document);
-            return ExtractCollectionName(context, originalCollectionName);
+            var originalCollectionName = CollectionName.GetCollectionName(document); // collection name from the document body (metadata)
+            var extractCollectionName = ExtractCollectionName(context, originalCollectionName); // collection name from table (or new)
+            
+            // Make the @collection prop in the @metadata be the same as is already saved in the collections table
+            if (extractCollectionName.Name.Equals(originalCollectionName) == false) 
+            { 
+                document.Modifications = new DynamicJsonValue
+                {
+                    [CollectionName.MetadataKeySegment] = new DynamicJsonValue
```

Comment:
This will delete all _existing_ metadata values.

Reply:
Moved to Studio